spec/octopus/serverspec/type/octopus_deploy_project_group_spec.rb

```diff
@@ -0,0 +1,112 @@
+    ex_acc_found_response = File.read('spec/octopus/serverspec/json/projectgroupsall.json')
+
+    it "handles projectgroup found" do
+      stub_request(:get, "https://octopus.example.com/api/").
+          to_return(status: 200, body: ex_does_not_support_spaces, headers: {})
+      stub_request(:get, "https://octopus.example.com/api/projectgroups/all?api-key=API-1234567890").
+          to_return(status: 200, body: ex_acc_found_response, headers: {})
+
+      wp = OctopusDeployProjectGroup.new("https://octopus.example.com", "API-1234567890", "Octopus Projects")
+      expect(wp.exists?).to be true
+    end
+
+    ex_acc_notfound_response = File.read('spec/octopus/serverspec/json/projectgroupsall.json')
+
+    it "handles projectgroup not found" do
+      stub_request(:get, "https://octopus2.example.com/api/").
+          to_return(status: 200, body: ex_does_not_support_spaces, headers: {})
+      stub_request(:get, "https://octopus2.example.com/api/projectgroups/all?api-key=API-0987654321").
+          to_return(status: 200, body: ex_acc_notfound_response, headers: {})
+
+      wp = OctopusDeployProjectGroup.new("https://octopus2.example.com", "API-0987654321", "Nonexistent projectgroup")
+      expect(wp.exists?).to be false
+    end
+
+    it "doesn't crash badly if handed a bad URL" do
+      stub_request(:get, "https://nonexistentdomain.com/api/").
+          to_return(status: 200, body: ex_does_not_support_spaces, headers: {})
+      stub_request(:get, "https://nonexistentdomain.com/api/projectgroups/all?api-key=API-1234567890").to_raise(SocketError)
+
+      expect { OctopusDeployProjectGroup.new("https://nonexistentdomain.com", "API-1234567890", "Octopus Projects") }.to raise_error(StandardError)
+    end
+
+  end
+
+  context "Server supports spaces" do
+
+    ex_supports_spaces = File.read('spec/octopus/serverspec/json/apisupportsspaces.json')
+    ex_projectgroup_found_response = File.read('spec/octopus/serverspec/json/projectgroupsall.json')
+
+    it "handles projectgroup found" do
+      stub_request(:get, "https://octopus.example.com/api/").
+          to_return(status: 200, body: ex_supports_spaces, headers: {})
+      stub_request(:get, "https://octopus.example.com/api/Spaces-1/projectgroups/all?api-key=API-1234567890").
+          to_return(status: 200, body: ex_projectgroup_found_response, headers: {})
+
+      wp = OctopusDeployProjectGroup.new("https://octopus.example.com", "API-1234567890", "Octopus Projects")
+
+      expect(wp.exists?).to be true
+    end
+
+    ex_projectgroup_notfound_response = File.read('spec/octopus/serverspec/json/projectgroupsall.json')
+
+    it "handles projectgroup not found" do
+      stub_request(:get, "https://octopus2.example.com/api/").
+          to_return(status: 200, body: ex_supports_spaces, headers: {})
+      stub_request(:get, "https://octopus2.example.com/api/Spaces-1/projectgroups/all?api-key=API-0987654321").
```

Comment:
Might be better to check against the non default space here...

Reply:
I was a bit doubtful about the value of that, but yeah, I see the point. changed.